packages/react-openapi/src/OpenAPIContextProvider.tsx

```diff
@@ -0,0 +1,72 @@
+'use client';
+
+import * as React from 'react';
+import { OpenAPIOperationData } from './fetchOpenAPIOperation';
+import { getServersURL } from './utils';
+
+type OpenAPIContextProps = {
+    isPending?: boolean;
+    serverUrl?: string;
+    state?: Record<string, string> | null;
+    onUpdate: (params: Record<string, string> | null) => void;
+};
+const OpenAPIContext = React.createContext<OpenAPIContextProps | null>(null);
+
+export function useOpenAPIContext() {
+    return React.useContext(OpenAPIContext);
+}
+
+export function OpenAPIContextProvider(props: {
+    children: React.ReactNode;
+    data: OpenAPIOperationData;
```

Comment:
Passing the `data` here goes against the whole server component approach. This means we'll include the OpenAPI operation data for every element in HTML, resulting in very big HTML output.

Reply:
I've refactored this a bit but could you explain the comment a bit more? 

The component is a client component that provides a React context, so I'm not sure how it affects the HTML output. But happy to admit it if I'm wrong as still getting a handle on some of the details of server components.